src/compiler/checker.ts

```diff
@@ -8450,6 +8453,12 @@ namespace ts {
             if ((isFunctionExpressionOrArrowFunction(func) || isObjectLiteralMethod(func)) &&
                 isContextSensitive(func) &&
                 func.kind !== SyntaxKind.ArrowFunction) {
+                const type = isObjectLiteralMethod(func)

```

Comment:
done, although it still does the work twice in the case that there is no contextualSignature.